記事一覧をAtomic Designに落とし込む

shikky0331 opened the pull request:
close #10

Commit:
refactor: extract ArticleTitle component as atom of Atomic Design

Commit:
refactor: extract Article component as molecules of Atomic Design

Commit:
refactor: extract ArticleList component as organisms of Atomic Design

Commit:
refactor: replace article list with components

piro0919 commented on src/components/organisms/ArticleList/index.tsx:
```diff
@@ -0,0 +1,20 @@
+const ArticleList: VFC<ArticleListProps> = ({
+  articles,
+}) => (
+  <>
```
[must]

リストという名前なので、HTML 的にもリストにしちゃったほうが良いかなーと思います。
あとからスタイルを追加したくなったときなどに、柔軟性が上がると思いますね。 👍

shikky0331 replied:
確かに名前と構造に剥離がありますね、、

[`9e84d5c` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/9e84d5c53a3a8c3d0b48ca1191f6b5311779b32e)
リスト構造に変更しました！

piro0919 commented on src/components/molecules/Article/style.module.scss:
```diff
@@ -0,0 +1,9 @@
+
+.post {
+  font-size: larger;
+  padding-top: 5em;
```
[must]

この padding は organisms ですね。 👀

shikky0331 replied:
[`d6b8d10` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/d6b8d10609902b9e82fc0003b4b5cb495841120d)
ArticleListで定義しました🙇‍♂️

shikky0331 replied:
[`afc1e8f` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/afc1e8fcc540e1d38ec7a1d8c344dc4980552115)
CSSを`li`タグに付与してしまい、globalなstyleになってしまっていました🙇‍♂️
修正しました

piro0919 commented on src/components/atoms/ArticleTitle/style.module.scss:
```diff
@@ -0,0 +1,4 @@
+.h3 {
+  margin-block-end: 0.5em;
```
[imo]

これちょっと微妙ですね…本来論としては atoms に持たせて良い margin ではないと思います。
きちんと設定するならここには 0 を設定して、上位のコンポーネントで間隔を設定してあげるべきだと思います。 👍

shikky0331 replied:
なるほどです！📝
どのComponentに`mergin`, `padding`を持たせるべきかの理解が曖昧なので、もう一度学び直したいと思います🙇‍♂️

[`d14d84b` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/d14d84b2fa7227e50ab80de244ea393d97ae5393)
修正しましたmm

piro0919 commented on src/components/templates/Blog/index.tsx:
```diff
@@ -1,38 +1,21 @@
 import styles from './style.module.scss';
 
-const Blog: VFC = () => (
+interface BlogProps {
```
[nits]

確か最近は interface でなく type のほうが推奨されていた気がしますね。
個人的には統一されていればどちらでも良いですが。 👍

piro0919 replied:
[must]

コンポーネント唯一の公開情報なので使用の有無に関わらず props の型には export をつけましょう！
逆にコンポーネントファイルが export して良い型は props だけですね。

shikky0331 replied:
[`7d63ae8` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/7d63ae8c7638bb9570ea0a0ee8258c867247c424)
こちらで`export`しました

shikky0331 replied:
> 確か最近は interface でなく type のほうが推奨されていた気がしますね。
> 個人的には統一されていればどちらでも良いですが。 👍

違いの理解が曖昧なので、調べたいと思います！

shikky0331 replied:
`Declaration merging`などの細かい違いはありますが、どちらを使っても大差はなさそうでした。
仕事のプロジェクトが`interface`をメインに使用しているので、`type`でいきたいと思います！

[`b377954` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/b3779548cd8db8123c84197987f8753349ea003f)

piro0919 commented on src/components/organisms/ArticleList/index.tsx:
```diff
@@ -0,0 +1,20 @@
+  articles,
+}) => (
+  <>
+    {articles.map(({
+      url, title, date, introduction,
+    }) => (
+      <Article url={url} title={title} date={date} introduction={introduction} />
```
[must]

key が設定されてないですね。 🙅‍♂️ 
別 issue を立てたので、先にこちらの対応をお願いしますー。 🙇 

rel https://github.com/shikky0331/shikky/issues/40

shikky0331 replied:
#41 でeslint-plugin-reactを追加しましたmm

[`e8de904` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/e8de904aa8ebc6b233701e4d331e64c6ef23a832)
keyを追加しました！

shikky0331 replied:
`key`を付与する場所が間違っていたので、修正いたしましたmm

[`39283c6` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/39283c65934700b76e83a8c2389d04d6f6b5ca87)

piro0919 commented on src/components/molecules/Article/index.tsx:
```diff
@@ -0,0 +1,23 @@
+  date,
+  introduction,
+}) => (
+  <article className={styles.post}>
```
[nits]

post って名前はちょっと微妙かと…article とか wrapper とかですかね。

shikky0331 replied:
しっかりと確認できてなかったです🙇‍♂️

[`d6b8d10` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/d6b8d10609902b9e82fc0003b4b5cb495841120d)
wrapperに修正しました！

piro0919 reviewed (changes requested):
[imo]

presentational components を切られる際は同時に story も切ってしまったほうが良いかなーと思いますが、どうでしょうか。
別 issue で対応される予定でしたら問題ないです。 👌

Commit:
perf: add key

Because performance is down

Commit:
feat: change to html list

Because match file name and html structure

Commit:
chore: change style name from post to wrapper

Make name easy to understand

Commit:
chore: migrate padding-top style from Article to ArticleList

Commit:
chore: migrate margin-block-end style from ArticleTitle to Article

Commit:
chore: export BlogProps interface

Commit:
fix: change li to .li style

Because li tag is global style

shikky0331 commented:
@piro0919 
https://github.com/shikky0331/shikky/pull/39#pullrequestreview-710121621

StoryBookも同じプルリクで実装します！mm

Commit:
chore: change interface to type

Because type and interface almost are same. I didn't use type much so I wanna use type.

Commit:
feat: add global styles to storybook

Commit:
feat: add MemoryRouter to storybook

Because use react-router in storybook

Commit:
fix: change location of key

Because location of key was wrong

Commit:
fix: change url in article

Use url as key of list so need to change url to different value

Commit:
feat: add ArticleTitle to storybook

Commit:
feat: add Article to storybook

Commit:
feat: add Article to storybook

shikky0331 commented:
@piro0919 
storybookを追加しました。お手隙の際に、ご確認をお願いします🙇‍♂️

piro0919 commented on .storybook/preview.js:
```diff
@@ -1,3 +1,7 @@
+import { addDecorator } from "@storybook/react";
+import { MemoryRouter } from "react-router";
```
[must]

react-router-dom でいけると思います。 👀 
以下で対応できないですかね？

`import { MemoryRouter as Router } from 'react-router-dom';`

shikky0331 replied:
react-router-dom で対応できました🙇‍♂️

[`c1b374a` (#39)](https://github.com/shikky0331/shikky/pull/39/commits/c1b374a071595441c86da702b3d7a7343a823be1)

Commit:
chore: change MemoryRouter from react-router to react-router-dom

piro0919 reviewed (approved):
LGTM 👍
